code/modules/integrated_electronics/core/special_pins/list_pin.dm

```diff
@@ -110,7 +110,7 @@
 /datum/integrated_io/lists/write_data_to_pin(var/new_data)
 	if(islist(new_data))
 		var/list/new_list = new_data
-		data = new_list.Copy(1,min( IC_MAX_LIST_LENGTH+1, new_list.len ))
+		data = new_list.Copy(max(1,new_list.len - IC_MAX_LIST_LENGTH+1),new_list.len +1)
```

Comment:
same as above

Reply:
yes,it's intentional for some reason.First option don't work.